src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -2835,9 +2844,13 @@ export class ResultsViewPageStore {
             this.mutations.isComplete &&
             this.mutationsByGene.isComplete
         ) {
-            return this.mutationMapperStoreByGene[gene.hugoGeneSymbol]
+            let store = this.mutationMapperStoreByGene[gene.hugoGeneSymbol]
                 ? this.mutationMapperStoreByGene[gene.hugoGeneSymbol]
                 : this.createMutationMapperStoreForSelectedGene(gene);
+
+            autorun(() => store.setSelectedTranscript(selectedTranscriptId));
```

Comment:
I know this is not a good way to handle it, but I got errors without `autorun`. Do you have better ideas on handling this?

![image](https://user-images.githubusercontent.com/16869603/84949441-3bfc8800-b0bb-11ea-8286-8e6b60270567.png)

Reply:
let me think about this for a bit.

Reply:
are we okay with an `autorun` here?

Reply:
@onursumer Do you have any suggestions? I spend a lot of time but didn't find a good solution... @alisman is helping me with it